### PR TITLE
fix #745 ValueError: random_int(): Argument #1 ($min) must be less than or equal to argument #2 ($max)

### LINKED ISSUE #745: ValueError: random_int(): Argument #1 ($min) must be less than or equal to argument #2 ($max)
NC 25.02
App 4.3.0

```
{"reqId":"gdwRHYwZLUGWfAsWcfsz","level":3,"time":"2022-12-09T10:40:02-06:00","remoteAddr":"","user":"--","app":"suspicious_login","method":"","url":"--","message":"Caught unknown error during IPv4 background training","userAgent":"--","version":"25.0.2.3","exception":{"Exception":"ValueError","Message":"random_int(): Argument #1 ($min) must be less than or equal to argument #2 ($max)","Code":0,"Trace":[{"file":"/var/www/nextcloud/apps/suspicious_login/lib/Service/NegativeSampleGenerator.php","line":74,"function":"random_int"},{"file":"/var/www/nextcloud/apps/suspicious_login/lib/Service/NegativeSampleGenerator.php","line":114,"function":"generateFromRealData","class":"OCA\\SuspiciousLogin\\Service\\NegativeSampleGenerator","type":"->"},{"function":"OCA\\SuspiciousLogin\\Service\\{closure}","class":"OCA\\SuspiciousLogin\\Service\\NegativeSampleGenerator","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"/var/www/nextcloud/apps/suspicious_login/lib/Service/NegativeSampleGenerator.php","line":115,"function":"array_map"},{"file":"/var/www/nextcloud/apps/suspicious_login/lib/Service/DataLoader.php","line":123,"function":"generateShuffledFromPositiveSamples","class":"OCA\\SuspiciousLogin\\Service\\NegativeSampleGenerator","type":"->"},{"file":"/var/www/nextcloud/apps/suspicious_login/lib/Service/TrainService.php","line":72,"function":"generateRandomShuffledData","class":"OCA\\SuspiciousLogin\\Service\\DataLoader","type":"->"},{"file":"/var/www/nextcloud/apps/suspicious_login/lib/BackgroundJob/TrainJobIpV4.php","line":70,"function":"train","class":"OCA\\SuspiciousLogin\\Service\\TrainService","type":"->"},{"file":"/var/www/nextcloud/lib/public/BackgroundJob/Job.php","line":78,"function":"run","class":"OCA\\SuspiciousLogin\\BackgroundJob\\TrainJobIpV4","type":"->"},{"file":"/var/www/nextcloud/lib/public/BackgroundJob/TimedJob.php","line":103,"function":"start","class":"OCP\\BackgroundJob\\Job","type":"->"},{"file":"/var/www/nextcloud/lib/public/BackgroundJob/TimedJob.php","line":93,"function":"start","class":"OCP\\BackgroundJob\\TimedJob","type":"->"},{"file":"/var/www/nextcloud/cron.php","line":152,"function":"execute","class":"OCP\\BackgroundJob\\TimedJob","type":"->"}],"File":"/var/www/nextcloud/apps/suspicious_login/lib/Service/NegativeSampleGenerator.php","Line":74,"message":"Caught unknown error during IPv4 background training","CustomMessage":"Caught unknown error during IPv4 background training"},"id":"6393826d60dee"}
```

### PR DESCRIPTION
**Pull request by furplag**

### CODE REVIEW

File: lib/Service/NegativeSampleGenerator.php [hunk 1]
```diff
@@ -71,7 +71,7 @@ private function getUniqueIPsPerUser(Dataset $positives): array {
 	private function generateFromRealData(array $uidVec, array $uniqueIps): array {
 		return array_merge(
 			$uidVec,
-			$uniqueIps[random_int(0, count($uniqueIps) - 1)]
+			empty($uniqueIps) ? array() : $uniqueIps[random_int(0, count($uniqueIps) - 1)]
```
Comment (ChristophWurst) [inline lib/Service/NegativeSampleGenerator.php:74]:
```suggestion
			empty($uniqueIps) ? [] : $uniqueIps[random_int(0, count($uniqueIps) - 1)]
```

**Review by ChristophWurst — COMMENTED** [review]
Coding style

**Review by ChristophWurst — APPROVED** [review]

**Comment by furplag** [comment]
Should I do something ?

**Comment by AndyXheli** [comment]
Fix for https://github.com/nextcloud/suspicious_login/issues/745

**Comment by furplag** [comment]
🙇

**Comment by joshtrichards** [comment]
Thanks, @furplag!

**Pull request merged**